BricksAndHearts.UnitTests/ControllerTests/Property/PropertyControllerTests.cs

```diff
@@ -0,0 +1,283 @@
+﻿using BricksAndHearts.Controllers;
+using BricksAndHearts.Database;
+using BricksAndHearts.Services;
+using BricksAndHearts.ViewModels;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using Xunit;
+
+namespace BricksAndHearts.UnitTests.ControllerTests.Property;
+
+public class PropertyControllerTests : PropertyControllerTestsBase
+{
+    [Fact]
+    public void AddNewPropertyContinueGet_CalledByUnregisteredUser_Returns403()
+    {
+        // Arrange 
+        CreateUnregisteredUserInController(_underTest);
+
+        // Act
+        var result = _underTest.AddNewProperty_Continue(1) as StatusCodeResult;
+
+        // Assert
+        result!.StatusCode.Should().Be(403);
+    }
+
+    [Fact]
+    public void AddNewPropertyBeginGet_ReturnsView_AtStep1()
+    {
+        // Arrange
+        CreateRegisteredUserInController(_underTest);
+
+        // Act
+        var result = _underTest.AddNewProperty_Begin() as ViewResult;
+
+        // Assert
+        result!.ViewName.Should().Be("AddNewProperty");
+        result.Model.Should().BeOfType<AddNewPropertyViewModel>().Which.Step.Should().Be(1);
+    }
+
+    [Fact]
+    public void AddNewPropertyContinueGet_ReturnsViewAtStep()

```

Comment:
If there are multiple steps, you could turn this test into a `[Theory]` - this allows you to parameterise the test so you can ensure it returns the view at step 2/3/4 as appropriate

Reply:
That's a great idea, thank you!